Tools/PlatformWPE.cmake

```diff
@@ -27,10 +27,11 @@ if (ENABLE_COG)
         set(WPE_COG_PKG_CONFIG_PATH ${CMAKE_BINARY_DIR})
     endif ()
     if ("${WPE_COG_REPO}" STREQUAL "")
-        set(WPE_COG_REPO "https://github.com/Igalia/cog.git")
+        # FIXME: Switch back to the upstream repo https://github.com/Igalia/cog.git.
+        set(WPE_COG_REPO "https://github.com/mcatanzaro/cog.git")
```

Comment:
I'm not very happy with the idea of changing the Cog upstream repo here.
Can we find a way of landing this on cog upstream disabled by default? so we can get Cog building both with the old API and also with the newer one? An idea to do that is that we can pass here to the Cog build system an optional build parameter similar to the one used here: `-DENABLE_2022_GLIB_API` that will allow in Cog to select using the newer api or the previous one.

Reply:
Does cog really need to support three different WPE API versions? Is it really a valuable thing to do? Yes it's certainly possible to do, but I figured two APIs would probably be enough? I did not remove the support for wpewebkit-1.0 and libsoup 2. I only removed wpewebkit-1.1 (libsoup 3 with old API). It's really not clear to me that we need to keep wpewebkit-1.1 in WebKit either; I suspect nothing uses it and it's just hanging around still because nobody has made the decision to remove it yet?

> I'm not very happy with the idea of changing the Cog upstream repo here.

If you give me access to Igalia/cog.git then I could land commits in a side branch there instead of having to use a fork. Alternatively, you could accept https://github.com/Igalia/cog/pull/524, but this would require some work to make the cog CI pass. Another alternative: retarget that pull request for a side branch and accept it into the side branch, so we can merge it to the upstream repo now, but in a side branch where we don't worry about fixing the CI. Then there's no longer any need to switch to a fork repo here and WebKit development can be unblocked.

BTW, we are running _very_ low on time to complete the API changes, so I don't think it makes sense to block on cog any longer tbh. I suggest we should return to cog next month _after_ API freeze begins on Feb 11. (We'll probably run past that deadline, but should try to get as much done before then as possible.)

Reply:
I guess we can also use a branch in cog repo and anybody can create a pr targeting that branch, no?

Reply:
> I guess we can also use a branch in cog repo and anybody can create a pr targeting that branch, no?

Yes. But I can't do it myself because I don't have permission. So please either do it, or give me permission so we can unblock this.

Reply:
Patrick created https://github.com/Igalia/cog/tree/mcatanzaro/next so that this pull request can proceed without changing the origin repo.